site/topics/sorting/sorting.rst

```diff
@@ -110,52 +110,64 @@ Selection Sort
     <iframe width="560" height="315" src="https://www.youtube.com/embed/ADD6jsSS9HI" frameborder="0" allowfullscreen></iframe>
 
 
-Insertion sort
+Insertion Sort
 ==============
 
-	.. image:: insertion.gif
+.. image:: insertion_sort.gif
+   :height: 333 px
+   :align: center
+   :target: https://en.wikipedia.org/wiki/Insertion_sort
 
-* You give me a list called ``in_list``
-* I create a new, empty, list called ``sorted_list``
-* For each element in ``in_list``
-    * I *insert* that element into ``sorted_list`` in the correct spot.
-    * e.g., if I'm asked to insert ``5`` into the list ``[1,3,7]``, I should end up with: ``[1,3,5,7]``
 
-.. admonition:: Activity
-    :class: activity
+* Insertion sort works by repeatedly inserting elements into their proper location within a collection
+* The high-level algorithm is as follows
 
-    Do an insertion sort, with pencil and paper, on the list ``[3,7,15,9,4,11,1,5,2]``. Record the value of ``sorted_list`` at each step.   
-   
-Let's have a look at an insertion sort implementation in Python::
+    * Start with the unsorted list and an empty list for the sorted elements
+    * For each element in the collection
 
-    def insertion_sort(in_list):
-        sorted_list = []
-        for element in in_list:
-            i = 0
-            while i < len(sorted_list) and (element > sorted_list[i]):
-                i = i + 1
-            sorted_list.insert(i, element)
-        return sorted_list
+        * Remove the element from the unsorted list
+        * Perform a linear search on the sorted list to find where the new element should be inserted
+        * Insert the new element into the sorted list
```

Comment:
Maybe worth clarifying _...at that index_ or something. Right now it feels incomplete, insert -- where?